common/JackRequest.h

```diff
@@ -1310,16 +1355,18 @@ struct JackSessionNotifyRequest : public JackRequest
     jack_session_event_type_t fEventType;
     int fRefNum;
 
-    JackSessionNotifyRequest()
+    JackSessionNotifyRequest() : fEventType(JackSessionSave), fRefNum(0)
     {}
     JackSessionNotifyRequest(int refnum, const char* path, jack_session_event_type_t type, const char* dst)
             : JackRequest(JackRequest::kSessionNotify), fEventType(type), fRefNum(refnum)
     {
+        memset(fPath, 0, sizeof(fPath));
+        memset(fDst, 0, sizeof(fDst));
         snprintf(fPath, sizeof(fPath), "%s", path);
+        fPath[JACK_MESSAGE_SIZE] = 0;
         if (dst) {
             snprintf(fDst, sizeof(fDst), "%s", dst);
-        } else {
-            fDst[0] = '\0';
+            fDst[JACK_CLIENT_NAME_SIZE] = 0;
```

Comment:
Is this semantically correct?

Old: terminating the char array at the first position
New: terminating the char array at the last (second to last?) position (some stuff is declared as SIZE+1)